python/common/org/python/stdlib/datetime/Date.java

```diff
@@ -0,0 +1,397 @@
+                throw new org.python.exceptions.SyntaxError("positional argument follows keyword argument");
+            }
+            if (kwargs.get("year") != null) {
+                this.year = kwargs.get("year");
+            } else {
+                this.year = args[0];
+            }
+
+            if (kwargs.get("month") != null) {
+                this.month = kwargs.get("month");
+            } else {
+                this.month = args[1];
+            }
+
+            if (kwargs.get("day") != null) {
+                this.day = kwargs.get("day");
+            } else {
+                this.day = args[2];
+            }
+
+            if ((this.year instanceof org.python.types.Int) && (this.month instanceof org.python.types.Int) && (this.day instanceof org.python.types.Int)) {
+                if (1 <= ((org.python.types.Int) this.year).value && ((org.python.types.Int) this.year).value <= 9999) {
+                    if (1d <= ((org.python.types.Int) this.month).value && ((org.python.types.Int) this.month).value <= 12d) {
+                        if (1d <= ((org.python.types.Int) this.day).value && ((org.python.types.Int) this.day).value <= 31d) {
+                        } else {
+                            throw new org.python.exceptions.ValueError("day is out of range for month");
+                        }
+                    } else {
+                        throw new org.python.exceptions.ValueError("month must be in 1..12");
+                    }
+                } else {
+                    throw new org.python.exceptions.ValueError("year " + this.year + " is out of range");
+                }
+            } else {
+                if (!(this.year instanceof org.python.types.Int)) {
+                    throw new org.python.exceptions.TypeError("integer argument expected, got " + this.year.typeName());
+                }
+                if (!(this.month instanceof org.python.types.Int)) {
+                    throw new org.python.exceptions.TypeError("integer argument expected, got " + this.month.typeName());
+                }
+                if (!(this.day instanceof org.python.types.Int)) {
+                    throw new org.python.exceptions.TypeError("integer argument expected, got " + this.day.typeName());
+                }
+            }
+        }
+
+        if (args.length + kwargs.size() == 2) {
+            if (args.length == 2) {
+                this.year = args[0];
+                this.month = args[1];
+            }
+
+            if (kwargs.get("year") != null) {
+                this.year = kwargs.get("year");
+            } else if (args.length > 0) {
+                this.year = args[0];
+            }
+
+            if (kwargs.get("month") != null) {
+                this.month = kwargs.get("month");
+            }
+            if (kwargs.get("day") != null) {
+                this.day = kwargs.get("day");
+            }
+
+            String y = this.year + "";
```

Comment:
Why turn these into strings instead of comparing to null directly?

Reply:
We also thought of this but since we didn't write the code we didn't change it.